core/src/main/scala/kafka/server/AddPartitionsToTxnManager.scala

```diff
@@ -33,7 +35,8 @@ object AddPartitionsToTxnManager {
 
 
 class TransactionDataAndCallbacks(val transactionData: AddPartitionsToTxnTransactionCollection,
-                                  val callbacks: mutable.Map[String, AddPartitionsToTxnManager.AppendCallback])
+                                  val callbacks: mutable.Map[String, AddPartitionsToTxnManager.AppendCallback],
+                                  val earliestAdditionMs: Long)
```

Comment:
The current approach I take is that we find the max verification time for the transaction data. I can either change this or clarify in the metric.

I wanted to take a per transaction level approach, but it gets a little bit harder to store the per transaction data since transaction data is an already defined collection. I could instead create a (parallel) map for each transaction if that makes sense.